src/gui/csvImport/CsvImportWidget.h

```diff
@@ -0,0 +1,78 @@
+﻿/*
+ *  Copyright (C) 2016 Enrico Mariotti <enricomariotti@yahoo.it>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEEPASSX_CSVIMPORTWIDGET_H
+#define KEEPASSX_CSVIMPORTWIDGET_H
+
+#include <QScopedPointer>
+#include <QPushButton>
+#include <QStringListModel>
+#include <QSignalMapper>
+#include <QList>
+#include <QComboBox>
+#include <QStackedWidget>
+
+#include "core/Metadata.h"
+#include "format/KeePass2Writer.h"
+#include "gui/csvImport/CsvParserModel.h"
+#include "keys/PasswordKey.h"
+
+
+namespace Ui {
+    class CsvImportWidget;
+}
+
+class CsvImportWidget : public QWidget
+{
+    Q_OBJECT
+
+public:
+    explicit CsvImportWidget(QWidget *parent = nullptr);
+    ~CsvImportWidget();
+    void load(const QString& filename, Database* const db);
+
+Q_SIGNALS:
```

Comment:
I think this can be merged as is. But please run a search and replace for Q_EMIT -> emit, Q_SIGNALS -> signals and Q_SLOTS -> slots over the source code before rebasing it. We migrated to MOC keywords recently.